### PR TITLE
MKL: Adding Batch Normalization implementation using the open source MKL-DNN Lib

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**

### CODE REVIEW

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc [hunk 1]
```diff
@@ -675,6 +681,627 @@ class MklFusedBatchNormGradOp : public OpKernel {
...
+        tf_shape_dst = src_tensor.shape();
+      }
+      AllocateOutputSetMklShape(context, dst_index, &dst_tensor,
+                                tf_shape_dst, dnn_shape_dst);
+
+      // Output of batchnorm has same shape as input.
+      dst.SetUsrMem(src_md, dst_tensor);
+
+      primitive bnrm_fwd_op;
+      if (is_training_)
```
Comment (andydavis1) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: Add braces around if/else here.

**Review by andydavis1 — APPROVED** [review]

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please.

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc (hunk 1 above)
Comment (claynerobison) [reply tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: Agreed. That's a pretty big single line of code.

**Comment by gunan** [comment]
Jenkins, test this please.
